### PR TITLE
Add AWS Kinesis Firehose Plugin

### PR DESCRIPTION
**Pull request by holidayworking**
I created AWS Kinesis Firehose Plugin.

### CODE REVIEW

**Comment by holidayworking** [comment]
Travis tests are failing.

```
mkdir -p build
for i in mackerel-plugin-*; do \
		go build  -ldflags="-s -w" -o build/$i \
		`pwd | sed -e "s|/home/travis/gopath/src/||"`/$i; \
	done


No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received

The build has been terminated
```

I have no idea……

**Comment by astj** [comment]
Maybe the failure is just an unlucky one. Now it passes 🙆

File: mackerel-plugin-aws-kinesis-firehose/lib/aws-kinesis-firehose.go [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+		Statistics: []*string{aws.String("Average")},
+		Namespace:  aws.String(namespace),
+	})
+	if err != nil {
+		return 0, err
+	}
+
+	datapoints := response.Datapoints
+	if len(datapoints) == 0 {
+		return 0, errors.New("fetched no datapoints")
```
Comment (astj) [inline mackerel-plugin-aws-kinesis-firehose/lib/aws-kinesis-firehose.go]: According to document, CloudWatch metrics of Firehose is collected every 5 minutes.
Therefore, this plugin may fetch no datapoints even if CloudWatch and plugin are working correctly.
Currently mackerel-agent outputs all logs written by plugins, so with this implementation, many "no datapoints" logs by this plugin will be written to mackerel-agent.log.
Do you have any idea to suppress this error?
(ex: mackerel-agent-aws-lambda say nothing about no datapoints: https://github.com/mackerelio/mackerel-agent-plugins/blob/master/mackerel-plugin-aws-lambda/lib/aws-lambda.go#L102-L104)


http://docs.aws.amazon.com/firehose/latest/dev/monitoring-with-cloudwatch-metrics.html
> The metrics that you configure for your Firehose delivery streams and agents are automatically collected and pushed to CloudWatch every five minutes.

**Comment by astj** [comment]
Thank you for submitting nice pull request. I reviewed!

File: mackerel-plugin-aws-kinesis-firehose/lib/aws-kinesis-firehose.go (hunk 1 above)
Comment (holidayworking) [inline mackerel-plugin-aws-kinesis-firehose/lib/aws-kinesis-firehose.go]: Thank you for the review.

> According to document, CloudWatch metrics of Firehose is collected every 5 minutes.
> Therefore, this plugin may fetch no datapoints even if CloudWatch and plugin are working correctly.

I overeloooked the document. I changed duration time. 

> Currently mackerel-agent outputs all logs written by plugins, so with this implementation, many "no datapoints" logs by this plugin will be written to mackerel-agent.log.
> Do you have any idea to suppress this error?

I suppressed "no datapoints" logs.

**Comment by astj** [comment]
LGTM. Thanks!!

**Comment by holidayworking** [comment]
Thank you!

**Pull request merged**